diffkemp/simpll/simpll.py

```diff
@@ -75,41 +86,86 @@ def simplify_modules_diff(first, second, fun_first, fun_second, var,
         first_out = LlvmKernelModule(first_out_name)
         second_out = LlvmKernelModule(second_out_name)
 
-        objects_to_compare = []
         missing_defs = None
-        syndiff_defs = None
         try:
+            graph = ComparisonGraph()
             simpll_result = yaml.safe_load(simpll_out)
             if simpll_result is not None:
-                if "diff-functions" in simpll_result:
-                    for fun_pair_yaml in simpll_result["diff-functions"]:
-                        fun_pair = [
-                            Result.Entity(
-                                fun["function"],
-                                fun["file"] if "file" in fun else "",
-                                fun["line"] if "line" in fun else None,
-                                "\n".join(
-                                    ["{} at {}:{}".format(call["function"],
-                                                          call["file"],
-                                                          call["line"])
-                                     for call in fun["callstack"]])
-                                if "callstack" in fun else "",
-                                fun["diff-kind"],
-                                fun["covered"]
-                            )
-                            for fun in [fun_pair_yaml["first"],
-                                        fun_pair_yaml["second"]]
-                        ]
-
-                        objects_to_compare.append(tuple(fun_pair))
+                if "function-results" in simpll_result:
+                    for fun_result in simpll_result["function-results"]:
+                        res_left = fun_result["first"]
+                        res_right = fun_result["second"]
+                        # Create the vertex and its edges.
+                        vertex = ComparisonGraph.Vertex(
+                            (res_left["function"], res_right["function"]),
+                            Result.Kind.from_string(fun_result["result"]),
+                            (res_left["file"] if "file" in res_left else None,
+                             res_right["file"] if "file" in res_right
+                             else None),
+                            (res_left["line"] if "line" in res_left
+                             else None,
+                             res_right["line"] if "line" in res_right
+                             else None)
+                        )
```

Comment:
The call is quite long, add explicit parameter names. Or, even better, make the `Vertex` constructor take `fun_result` as a parameter and parse it inside the constructor.

Reply:
The idea was to keep the YAML parsing in `simpll.py`, keeping only code related to the graph logic in `caching.py` (this also applies to other calls in the function).